srcCpp03/RTIDDSImpl.cxx

```diff
@@ -557,13 +609,111 @@ class RTIPublisher: public RTIPublisherBase<T> {
         if (!isCftWildCardKey) {
             this->_writer.write(this->data, this->_instance_handles[key]);
         } else {
-            this->_writer.write(this->data,
-                    this->_instance_handles[this->_num_instances]);
+            this->_writer.write(this->data, this->getCftInstanceHandle());
         }
         return true;
     }
 };
 
+#ifdef RTI_FLATDATA_AVAILABLE
```

Comment:
Add a comment explaining a bit about this class, its use and why we use it differently.

Reply:
Done for every class and method of new FlatData classes